public/directives/todoDetail/todoDetailDirective.js

```diff
@@ -1,16 +1,46 @@
-angular.module('app').directive('todoDetail', function() {
+angular.module('app').directive('todoDetail', function () {
 
-    return {
-        templateUrl: './directives/todoDetail/todoDetailTemplate.html',
-        replace: true,
-        controller: function($scope) {
+  return {
+    templateUrl: './directives/todoDetail/todoDetailTemplate.html',
+    replace: true,
+    controller: function ($scope) {
+      $scope.changesMade = false;
+      var originalVals = null;
 
-        },
-        scope: {
-            task: '<',
-            onSave: '&',
-            // onRestore: '&',
-            // onDelete: '&'
+      function resetValues() {
+        $scope.changesMade = false;
+        originalVals = angular.copy($scope.task);
+      }
+
+      $scope.$watch('task', function(newVal, oldVal) {
+        if(!angular.equals(newVal, oldVal)) {
+          originalVals = angular.copy(newVal);
+        }
+      });
+
+      $scope.$watchCollection('task', function (newVal) {
+        if (newVal) {
+          if (!originalVals) originalVals = angular.copy(newVal);
+          $scope.changesMade = !angular.equals(originalVals, newVal);
         }
+      });
```

Comment:
`$scope.$watchCollection` is similar to `$scope.$watch`, but it watches the values one layer deep (in this case it will watch the `title` and `notes` properties for changes because those are one layer deep into the task object. I did this so that I could watch for changes on the inputs and be able to show and hide the `save` and `cancel` buttons accordingly. Since we have two input fields, I needed to make sure any change that happened in either one or both would be monitored.